lightning/src/util/events.rs

```diff
@@ -118,6 +118,8 @@ pub enum Event {
 		/// Note that this serves as a payment receipt, if you wish to have such a thing, you must
 		/// store it somehow!
 		payment_preimage: PaymentPreimage,
+		/// The hash which was given to ChannelManager::send_payment.
```

Comment:
This should be a link (ie wrapped in [\`  and `]). Also, should mention when its `None` and when its not (reading old events)

Reply:
I've extended the comment. See the latest update.

Reply:
Links are resolved locally, so you need to provide a full path at the end which tells `cargo doc` exactly where to look, eg:

```
--- a/lightning/src/util/events.rs
+++ b/lightning/src/util/events.rs
@@ -120,5 +120,7 @@ pub enum Event {
                payment_preimage: PaymentPreimage,
-               /// The hash which was given to [ChannelManager::send_payment].
+               /// The hash which was given to [`ChannelManager::send_payment`].
                /// When the payment sent message comes from an older library version,
                /// the payment hash will be missing, in which case the value of this field will be `None`.
+               ///
+               /// [`ChannelManager::send_payment`]: crate::ln::channelmanager::ChannelManager::send_payment
                payment_hash: Option<PaymentHash>,
```